### ISSUE TITLE
when embedded in spring boot, hawt.io should use the management settings

### ISSUE THREAD

**Issue by NetForce1** [hawtio/hawtio#1972]
Spring Boot can be configured to use separate host, port and/or context path for the management utilities. When hawt.io is running embedded within a Spring Boot app, it should honor these settings.

**Comment by davsclaus** [comment]
Can you be more specific? Do you mean that hawtio web console context-path is hardcoded to /hawtio or something and you want it to read those management settings and use a dynamic computed context path?

In other words a bit more details would help us better understand.

**Comment by NetForce1** [comment]
Spring Boot supports binding management endpoints to host, port, context that is different from the main host, port, context. That gives you the opportunity to only bind management endpoints on an internal port that is not exposed through the firewall.

See this section in the reference guide: http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#production-ready-monitoring

**Comment by alwaysastudent** [comment]
+1

The fix is easy HawtioConfiguration should be annotated with `@ManagementContextConfiguration` instead of `@Configuration` and add that in their `spring.factories`

I just worked around this easily by doing the below
```
@ManagementContextConfiguration
public class CustomHawtioConfiguration extends HawtioConfiguration {
....
}
```
And configured the same on our `META-INF/spring.factories`
```
#Management Context Configuration
org.springframework.boot.actuate.autoconfigure.ManagementContextConfiguration=\
my.package.CustomHawtioConfiguration
```
That is should do it.

**Comment by davsclaus** [comment]
Yay sounds good @alwaysastudent. Wonder if you can create a PR with your fix?

**Comment by alwaysastudent** [comment]
Yes, can I do this over this week or early next ?

**Comment by davsclaus** [comment]
Yes sound good

**Comment by jeacott1** [comment]
still open a year later :(

**Comment by evmin** [comment]
+1

**Comment by vspiliop** [comment]
+1

**Comment by tadayosi** [comment]
I believe I've finally fixed it. The one remaining limitation is the `management.context-path` property still doesn't work as expected. Let's fix it at a separate issue #2138 and close this one.
